src/textual/pilot.py

```diff
@@ -104,21 +112,41 @@ async def click(
         else:
             target_widget = screen
 
+        if not target_widget.size.contains(*offset):
```

Comment:
I don't understand this. The offset should be added to the click coordinate. 

It is not an error to have an offset that puts the click position outside of the selected widget. It is only out of bounds if the click is outside of the screen.

Reply:
Refactored after the back-and-forth in the original issue.